dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/AbstractOracleConnector.java

```diff
@@ -156,9 +157,20 @@ String getFormatName() {
   @Override
   public Handle open(@Nonnull ConnectorArguments arguments) throws Exception {
     Driver driver = newDriver(arguments.getDriverPaths(), "oracle.jdbc.OracleDriver");
-    DataSource dataSource =
-        new SimpleDriverDataSource(driver, buildUrl(arguments), buildProperties(arguments));
-    return new JdbcHandle(dataSource);
+    String url = buildUrl(arguments);
+    DataSource dataSource = new SimpleDriverDataSource(driver, url, buildProperties(arguments));
+    try {
+      return new JdbcHandle(dataSource);
+    } catch (Exception e) {
```

Comment:
The primary message that we want to fix is
```
* ERROR
* IO Error: Got minus one from a read call
* Caused by: Got minus one from a read call
```

The fix is good, but it needs to be ensured that other errors are unaffected. So I'd make the `catch` clause as narrow as possible to avoid catching anything else.

Reply:
makes sense, thanks, changed it to `SQLRecoverableException`